File: frontend/src/api/execOverview.ts
/**
 * Executive Overview API — министерский обзор сектор→компания→проекты+дедлайны.
 * Backend: /backend/app/api/routes/exec_overview.py
 */
import { api } from "./client";

export type DeadlineState = "overdue" | "month" | "quarter" | "later" | "none";

export interface ExecOverviewProject {
  id: string;
  title: string;
  description: string | null;
  direction: string | null;
  direction_id: string | null;
  status: string;
  progress_percent: number;
  due_date: string | null;
  deadline_state: DeadlineState;
}

export interface ExecOverviewDirection {
  id: string;
  code: string;
  name: string;
}

export interface ExecOverviewCompany {
  id: string;
  code: string;
  name: string;
  total: number;
  overdue: number;
  revenue: number | null;
  profit: number | null;
  fin_year: number | null;
  projects: ExecOverviewProject[];
}

export interface ExecOverviewSector {
  id: string | null;
  code: string | null;
  name: string;
  color: string | null;
  short_badge: string | null;
  total: number;
  overdue: number;
  company_count: number;
  companies: ExecOverviewCompany[];
}

export interface ExecOverviewResponse {
  year: number | null;
  as_of: string;
  total: number;
  overdue: number;
  due_this_month: number;
  sector_count: number;
  company_count: number;
  sectors: ExecOverviewSector[];
  directions: ExecOverviewDirection[];
}

export const execOverviewApi = {
  get(year?: number | null): Promise<ExecOverviewResponse> {
    const q = year != null ? `?year=${year}` : "";
    return api.get<ExecOverviewResponse>(`/exec-overview${q}`).then(r => r.data);
  },
};

File: backend/app/schemas/exec_overview.py
"""Executive Overview (министерский обзор) — иерархия сектор→компания→проект.

Один лёгкий агрегат для министра: по каждому сектору его компании и их ТЕКУЩИЕ
проекты с дедлайнами, направлением и кратким описанием. Без тяжёлых метрик.
"""
from __future__ import annotations

from datetime import date
from typing import Literal, Optional
from uuid import UUID

from pydantic import BaseModel, Field

DeadlineState = Literal["overdue", "month", "quarter", "later", "none"]


class ExecOverviewProject(BaseModel):
    id: UUID
    title: str
    description: Optional[str] = None
    direction: Optional[str] = None
    direction_id: Optional[UUID] = None
    status: str
    progress_percent: int = 0
    due_date: Optional[date] = None
    deadline_state: str = "none"
    # «Ход проекта» — последний нарративный апдейт (status_update)
    last_update: Optional[str] = None
    last_update_at: Optional[date] = None
    last_update_health: Optional[str] = None
    last_update_author: Optional[str] = None


class ExecOverviewDirection(BaseModel):
    """Каталог стратегических направлений платформы (для дорожной карты)."""
    id: UUID
    code: str
    name: str


class ExecOverviewTask(BaseModel):
    """Задача проекта (для разворота по клику)."""
    id: UUID
    title: str
    status: str
    assignee_name: Optional[str] = None
    progress_percent: int = 0
    due_date: Optional[date] = None
    deadline_state: str = "none"


class ExecOverviewCompany(BaseModel):
    id: UUID
    code: str
    name: str
    total: int = 0
    overdue: int = 0
    # Финпоказатели (последний доступный год; абс. UZS) — оставлены для совместимости
    revenue: Optional[float] = None
    profit: Optional[float] = None
    fin_year: Optional[int] = None
    # Ключевые результаты бизнес-плана за Q1 (план/факт, абс. UZS) — gated bp.view
    q1_revenue_plan: Optional[float] = None
    q1_revenue_fact: Optional[float] = None
    q1_profit_plan: Optional[float] = None
    q1_profit_fact: Optional[float] = None
    projects: list[ExecOverviewProject] = Field(default_factory=list)


class ExecOverviewSector(BaseModel):
    id: Optional[UUID] = None
    code: Optional[str] = None
    name: str
    color: Optional[str] = None
    short_badge: Optional[str] = None
    total: int = 0
    overdue: int = 0
    company_count: int = 0
    companies: list[ExecOverviewCompany] = Field(default_factory=list)


class ExecOverviewResponse(BaseModel):
    year: Optional[int] = None
    as_of: date
    total: int = 0
    overdue: int = 0
    due_this_month: int = 0
    sector_count: int = 0
    company_count: int = 0
    sectors: list[ExecOverviewSector] = Field(default_factory=list)
    directions: list[ExecOverviewDirection] = Field(default_factory=list)
